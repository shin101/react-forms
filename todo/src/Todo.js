import React from "react";

function Todo ({name, id, remove }) {
  const handleDelete = () => remove(id);
 
  return (
    <div>
      <ul>
        <li>{name}</li>
        <button onClick={handleDelete}>X</button>
      </ul>
    </div>
  )

};

export default Todo;