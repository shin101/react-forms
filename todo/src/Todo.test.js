import React from "react";
import Todo from "./Todo";
import { render, fireEvent } from "@testing-library/react";

// smoke test
it("renders without crashing", function(){
  render(<Todo />);
});

// snapshot test
it("matches snapshot", function (){
  const { asFragment } = render(<Todo />);
  expect(asFragment()).toMatchSnapshot();
})


  // see if delete button works correctly
  it("deletes a task correctly", function (){
    const mockRemove = jest.fn();
    const { queryByText } = render(<Todo remove={mockRemove}/>);
    // eslint-disable-next-line
    const btn = queryByText("X");
    fireEvent.click(btn);
    expect(mockRemove).toHaveBeenCalled();
  })

