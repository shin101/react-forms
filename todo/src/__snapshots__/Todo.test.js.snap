// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <ul>
      <li />
      <button>
        X
      </button>
    </ul>
  </div>
</DocumentFragment>
`;
